### DISCUSSION TITLE
I think that Tip in the FastAPI documentation is incorrect: Virtual Environments: Activate the Virtual Environment

### DISCUSSION THREAD

**Discussion by alv2017** [fastapi/fastapi#13021] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Issue with the docs, hence no code provided
```

### Description

* Open documentation page Virtual Environments, Activate The Virtual Environment section [Link](https://fastapi.tiangolo.com/virtual-environments/#activate-the-virtual-environment)
* Tip Text:

```
Every time you install a new package in that environment, activate the environment again.


This makes sure that if you use a terminal (CLI) program installed by that package, you use the 

one from your virtual environment and not any other that could be installed globally, probably

with a different version than what you need.
```

* The tip statement is misleading:

1)  when one installs the package into the virtual environment, there is no need to re-activate it
2) when the virtual environment is activated the global packages are not visible to it, this is the main idea of using venvs. 

### Operating System

Other

### Operating System Details

This is a documentation issue, and it doesn't relate to operation systems

### FastAPI Version

0.115.5

### Pydantic Version

2.9.2

### Python Version

3.12

### Additional Context

Documentation issue: Virtual Environments: Activate the Virtual Environment

**Comment by AliYmn** [comment, marked as answer]
I agree with your assessment - the tip appears to be incorrect on both counts:

1. You're right that there's no need to reactivate a virtual environment after installing new packages. Once activated, the virtual environment maintains its state and will recognize newly installed packages immediately.

2. And yes, the second point is also correct - isolation from global packages is indeed one of the fundamental features of virtual environments. When a virtual environment is activated, it automatically uses its package directory, which is utterly separate from global packages.

Would I help you submit this as a documentation improvement PR? The correction would make the docs more accurate for other users.

The tip could potentially be removed entirely or replaced with something more useful like:

"Make sure your virtual environment is activated before installing packages to ensure they're installed in the virtual environment rather than globally."

**Reply by alv2017** [reply]
I think that it would be the best to remove this tip. The shorter documentation is the better. :smile:

**Comment by alv2017** [comment]
@AliYmn I created a new PR that removes this misleading tip: #13025 

Couldn't you please have a look? (Thank you :smile:)
